packages/at_secondary_server/lib/src/server/at_secondary_impl.dart

```diff
@@ -172,7 +173,29 @@ class AtSecondaryServerImpl implements AtSecondaryServer {
         .getSecondaryPersistenceStore(currentAtSign)!;
 
     // Initialize enrollment manager
-    enrollmentManager = EnrollmentManager(secondaryKeyStore);
+    enrollmentManager = EnrollmentManager(secondaryKeyStore, currentAtSign);
+    List<String> deletedKeys =
+        await enrollmentManager.removeLegacyApkamPublicKeys();
+    if (deletedKeys.isNotEmpty) {
+      logger.info('Removed legacy APKAM public keys: $deletedKeys');
+    }
+    deletedKeys = await enrollmentManager.removeOrphanedApkamEncryptionKeys();
+    if (deletedKeys.isNotEmpty) {
+      logger.info('Removed orphaned APKAM encryption keys: $deletedKeys');
+    }
```

Comment:
The code above (1) removes legacy APKAM public keys which we do not need and (2) removes APKAM encryption keys which were orphaned after APKAM expiries (part of this PR ensures that there are no more orphans in the future)